src/main.rs

```diff
@@ -535,79 +535,80 @@ impl ManParser for Type1 {
         manpage.contains(r#".SH "OPTIONS""#)
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
+    fn parse_man_page(&mut self, manpage: &str, cmd_name: &str) -> Option<String> {
+        let options_section_re = regex!(r#"\.SH "OPTIONS"((?s:.)*?)(\.SH|\z)"#);
+        let options_section_matched = options_section_re.find(manpage);
+
+        let mut options_section = options_section_matched.unwrap().as_str();
+        let options_parts_re = regex!(r"\.PP((?s:.)*?)\.RE");
+        let mut options_matched = options_parts_re.captures(options_section);
+        // add_diagnostic(format!("Command is {}", cmd_name));
+
+        if options_matched.is_none() {
+            // add_diagnostic("Unable to find options");
+            return self
+                .fallback(options_section, cmd_name)
+                .or_else(|| self.fallback2(options_section, cmd_name));
+        }
+
+        let mut built_command_output = Vec::new();
+        let mut existing_options = HashSet::new();
+        while let Some(mat) = options_matched {
+            let mut data = mat.get(1).unwrap().as_str();
+            let last_dotpp_index = data.rfind(".PP");
+            if let Some(idx) = last_dotpp_index {
+                data = &data[idx + 3..];
+            }
+
+            let data = remove_groff_formatting(data);
+            let data: Vec<_> = data.split(".RS 4").collect();
+            if data.len() > 1 {
+                let option_name = data[0].trim();
+                if option_name.find('-').is_some() {
```

Comment:
Why not
```suggestion
                if option_name.contains('-') {
```

Reply:
Thanks, nice findings. I forgot about that.